improver/calibration/reliability_calibration.py

```diff
@@ -1350,19 +1359,16 @@ def _apply_point_by_point_calibration(
         y_name = forecast.coord(axis="y").name()
         x_name = forecast.coord(axis="x").name()
 
-        # create list of dimensions other than time dimension
-        # and threshold dimension
+        # create list of dimensions
         dim_names = get_dim_coord_names(forecast)
         dim_associated_coords = {}
 
-        # create lists with first entry a dimension name and second
-        # entry a list of auxiliary coordinates associated with that
-        # dimension. Collect these lists together into a list.
-        for dim_name in dim_names:
-            template_dim = [x for x in forecast.coord_dims(dim_name)]
+        # create dictionary with dimension name as keys, containing auxiliary
+        # coordinates associated with that dimension.
+        for dim_index, dim_name in enumerate(dim_names):
             associated_coords = [
                 c
-                for d in template_dim
+                for d in [dim_index]
```

Comment:
This for loop isn't needed anymore as dim_index is now just a number so you can set `dimensions=dim_index`

Reply:
Good point, changed to `associated_coords = [c for c in forecast.coords(dimensions=dim_index, dim_coords=False)]`